tests/admin_views/test_multidb.py

```diff
@@ -0,0 +1,61 @@
+from unittest.mock import patch
+
+from django.conf.urls import url
+from django.contrib import admin
+from django.contrib.auth.models import User
+from django.db import connections, router
+from django.test import TestCase, override_settings
+
+from .models import Book
+
+
+class Router(object):
+    def __init__(self, target):
+        self.target = target
+
+    def db_for_read(self, model, **hints):
+        return self.target
+
+    def db_for_write(self, model, **hints):
+        return self.target
+
+site = admin.AdminSite(name="test_adminsite")
+site.register(Book)
+
+urlpatterns = [
+    url(r'^admin/', site.urls),
+]
+
+
+@override_settings(ROOT_URLCONF=__name__)
+class MultiDatabaseTest(TestCase):
+    multi_db = True
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.original_routers = router.routers
+        cls.superuser = {}
+        for db in connections:
+            router.routers = [Router(db)]
+            cls.superuser[db] = User.objects.create_superuser(username='admin', password='something',
+                                                              email='test@test.org')
+
+    @classmethod
+    def tearDownClass(cls):
+        router.routers = cls.original_routers
+        return super().tearDownClass()
+
+    @patch('django.contrib.admin.options.transaction')
+    def test_save_to_each_db(self, mock):
+        for db in connections:
+            # The router must be set before we login.
+            router.routers = [Router(db)]
```

Comment:
Or use the `self.settings()` context manager if you can't decorate the class since you have a loop.